services/management/ia/templates_service.go

```diff
@@ -203,17 +341,80 @@ func (svc *TemplatesService) ListTemplates(ctx context.Context, req *iav1beta1.L
 
 // CreateTemplate creates a new template.
 func (svc *TemplatesService) CreateTemplate(ctx context.Context, req *iav1beta1.CreateTemplateRequest) (*iav1beta1.CreateTemplateResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method CreateTemplate not implemented")
+	pParams := &alert.ParseParams{
+		DisallowUnknownFields: true,
+		DisallowInvalidRules:  true,
+	}
+
+	rules, err := alert.Parse(strings.NewReader(req.Yaml), pParams)
+	if err != nil {
+		svc.l.Errorf("failed to parse rule template form request: +%v", err)
+		return nil, status.Error(codes.InvalidArgument, "Failed to parse rule template.")
+	}
+
+	if len(rules) != 1 {
+		return nil, status.Error(codes.InvalidArgument, "Request should contain exactly one rule template.")
+	}
+
+	params := &models.CreateTemplateParams{
+		Rule:   &rules[0],
+		Source: iav1beta1.TemplateSource_USER_API.String(),
+	}
+
+	e := svc.db.InTransaction(func(tx *reform.TX) error {
+		var err error
+		_, err = models.CreateTemplate(tx.Querier, params)
+		return err
+	})
+	if e != nil {
+		return nil, e
+	}
+
+	return &iav1beta1.CreateTemplateResponse{}, nil
 }
 
 // UpdateTemplate updates existing template, previously created via API.
 func (svc *TemplatesService) UpdateTemplate(ctx context.Context, req *iav1beta1.UpdateTemplateRequest) (*iav1beta1.UpdateTemplateResponse, error) {
```

Comment:
Should we after Create/Update refresh in-memory state, or it's up to the user (`reload` parameter in `ListTemplates` request)